### PR TITLE
#186 Adding ztsroletoken library

### PR DESCRIPTION
**Pull request by psasidhar**

### CODE REVIEW

File: libs/go/ztsroletoken/Makefile [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#
+# Makefile to build ZTS Role Token client utility
+# Prerequisite: Go development environment
+#
+# Copyright 2016 Yahoo Inc.
```
Comment (havetisyan) [inline libs/go/ztsroletoken/Makefile]: should be 2017 Yahoo Holdings, Inc

please make the change in all other files as well.

File: libs/go/ztsroletoken/role-token.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
+package roletoken
+
+import (
+	"sync"
```
Comment (havetisyan) [inline libs/go/ztsroletoken/role-token.go]: aren't we running the import check on these files? they seem to be out of order

File: libs/go/ztsroletoken/role-token.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
+package roletoken
```
Comment (havetisyan) [inline libs/go/ztsroletoken/role-token.go]: add license lines:

// Copyright 2017 Yahoo Holdings, Inc.
// Licensed under the terms of the Apache version 2.0 license. See LICENSE file for terms.

File: libs/go/ztsroletoken/pom.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright 2016 Yahoo Inc.
```
Comment (havetisyan) [inline libs/go/ztsroletoken/pom.xml]: fix copyright line

File: libs/go/ztsroletoken/role-token.go [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+	"crypto/x509"
+	"fmt"
+
+	"github.com/yahoo/athenz/clients/go/zts"
+	"github.com/yahoo/athenz/libs/go/zmssvctoken"
+)
+
+const (
+	defaultPrincipalAuthHeader = "Athenz-Principal-Auth"
```
Comment (havetisyan) [inline libs/go/ztsroletoken/role-token.go]: Having this as a const, will it prohibit the use in a different environment with a different header name?

File: libs/go/ztsroletoken/role-token_test.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
+package roletoken
```
Comment (havetisyan) [inline libs/go/ztsroletoken/role-token_test.go]: please add license info

Comment (psasidhar) [inline libs/go/ztsroletoken/role-token_test.go]: done

File: libs/go/ztsroletoken/role-token.go (hunk 5 above)
Comment (psasidhar) [inline libs/go/ztsroletoken/role-token.go]: User can pass the custom header through RoleTokenOptions.AuthHeader member

File: libs/go/ztsroletoken/role-token.go (hunk 2 above)
Comment (psasidhar) [inline libs/go/ztsroletoken/role-token.go]: fixed the formatting

**Pull request merged**